libnopegl/src/node_animkeyframe.c

```diff
@@ -319,6 +323,175 @@ static easing_type back_derivative(easing_type t, easing_type s)
 DECLARE_EASINGS(back,            , back_func(x, PARAM(0, 1.70158)), TRANSFORM)
 DECLARE_EASINGS(back, _derivative, back_derivative(x, PARAM(0, 1.70158)), DERIVATIVE)
 
+/* Cubic Bezier */
+
+static int double_close_to(double value, double ref, double eps)
+{
+    return fabs(value - ref) < eps;
+}
+
+static float clamp_root(float r)
+{
+    const float s = NGLI_CLAMP(r, 0.f, 1.f);
+    return fabsf(s - r) > FLT_EPSILON ? NAN : s;
```

Comment:
`FLT_EPSILON` is, as far as I can tell, not a very good threshold choice as it's probably going to filter out all sorts of cases where the root is close to 0 or 1: it's a *very* restrictive value, and at the same time not 100%. A more arbitrary 1e-6 (that can be understood as a "precision" rounding) will likely do a better job.